src/llfs/trace_refs_recursive.hpp

```diff
@@ -59,6 +68,19 @@ inline batt::Status trace_refs_recursive(PageTracer& page_tracer, IdTypePairSeq&
   return batt::OkStatus();
 }
 
+/** \brief Convenience overload; creates a LoadingPageTracer to wrap the page_loader arg, then calls
+ * the PageTracer variant of trace_refs_recursive.
+ */
+template <typename IdTypePairSeq, typename Pred, typename Fn>
+inline batt::Status trace_refs_recursive(PageLoader& page_loader, IdTypePairSeq&& roots,
```

Comment:
Do you think it'd be useful to have another convenience overload that takes in a `PageLoader` (as this one already does) and also a `const std::vector<std::unique_ptr<PageDeviceEntry>>&` as a parameter so we can have an overload that can create a `CachingPageTracer` too?

Reply:
In this case I don't think its worth it.  The major reason to create the PageLoader overload is to avoid breaking API compatibility with the previous LLFS release (or rather, to fix the breakage that was introduced).  FWIW, in my experience it is often better to wait until you know you need such an API to add it, instead of adding them speculatively.

Reply:
Makes sense, thanks for the explanation!